### ISSUE TITLE
bug(mat-optgroup): <label> element should be skipped during screen-reader page navigation

### ISSUE THREAD

**Issue by axverma** [angular/components#21857]
#### Reproduction

1. Go to the "Creating groups of options" demo on the [mat-select demo page](https://material.angular.io/components/select/overview#creating-groups-of-options).
2. Navigate through the mat-select with groups using a screen-reader's page navigation. 
 
#### Expected Behavior

The group label should only be read once when the screen-reader navigates through the select options. The group label itself should never get focus, but its value should be read when the user navigates into the group.  

#### Actual Behavior

ChromeVox (issue): The screen-reader will read focus the group label and read the group name twice ("Grass" followed by "Grass group")

MacOS VoiceOver + Chrome (issue): The screen-reader focuses the group as a whole and reads the group name ("Grass, group 2 of 5"). Then it progresses to the group label ("Grass") before going into the options.

Windows JAWS + Chrome (ok): The screen-reader skips the label element and reads the group label when navigating into a group of options.

#### Guidance

mat-select is a listbox widget, and per the [WAI ARIA listbox guidance](https://www.w3.org/TR/wai-aria-1.2/#listbox), listboxes should "contain children whose role is option or elements whose role is group which in turn contains children whose role is option." The `<label>` elements interfere with this pattern because they are not hidden from the a11y tree. 

We can fix this by adding `aria-hidden="true"` to the `<label>` element, so that while screenreaders will still announce the label text for the group (due to the aria-labelledby property on mat-optgroup), they will skip over the actual label element. Optionally, the `<label>` could be changed to a `<span>` to make it clear that the text is visual only and not carrying any special semantic meaning for the a11y tree.
